### PR TITLE
hw pop-up

### PR DESCRIPTION
**Pull request by Zihfred**
### HW POP-UP

[Demo](https://zihfred.github.io/Pop-up/) |
[Code base](https://github.com/Zihfred/Pop-up)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Zihfred/google pop-up/index.html [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            <a href="#"><p>Gmail</p></a>
+        </div>
+        <div class="menu_item">
+            <a href="#"><p>Images</p></a>
+        </div>
+
+        <div class="menu_item pop-up-click">
+            <input type="checkbox" class="popup-checkbox" id="check">
+            <label for="check"><i class="fas fa-th"></i></a></label>
+            <div class="pop-up" id="popup">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: Try to keep one style in class/id name spacing.

File: submissions/Zihfred/google pop-up/index.html [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    <div class="pop-up-item close">
+                        <img src="img/Без названия (1).jpg" class="pop-up-item__img" alt="">
+                        <p class="pop-up-item__text">Contacts</p>
+                    </div>
+                    <div class="pop-up-item close">
+                        <img src="img/Hangouts_Icon.png" class="pop-up-item__img" alt="">
+                        <p class="pop-up-item__text">Hangouts</p>
+                    </div>
+                    <div class="pop-up-item close">
+                        <img src="img/Без названия (1).png" class="pop-up-item__img" alt="">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: Better avoid file name like this.

File: submissions/Zihfred/google pop-up/index.html [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        <div class="menu_item pop-up-click">
+            <input type="checkbox" class="popup-checkbox" id="check">
+            <label for="check"><i class="fas fa-th"></i></a></label>
+            <div class="pop-up" id="popup">
+                <input type="checkbox" id="more-сlick">
+                <div class="pop-up-content">
+                    <input type="checkbox" id="more-checkbox">
+                    <div class="pop-up-item">
+                        <img src="img/google_docs.png" class="pop-up-item__img" alt="">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: Fill `alt` attribute on `img`.

File: submissions/Zihfred/google pop-up/index.html [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        </div>
+        <div class="menu_item username">
+            <a href="#"><p>A</p></a>
+        </div>
+
+
+    </div>
+</div>
+</body>
+</html>
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: Add an empty line at the end of every file.

File: submissions/Zihfred/google pop-up/style.css [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    display: inline-block;
+    position: relative;
+    content: "";
+    top: -9px;
+    right: -216px;
+    margin-left: 4px;
+    width: 10px;
+    height: 10px;
+    transform: rotate(45deg);
+    border-left: 1px solid rgba(0,0,0,.2);
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: In the previous line, you use `0.75` format for float numbers here you use `.2` Try to keep one code style.

File: submissions/Zihfred/google pop-up/style.css [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    opacity: 0;
+    display: none;
+}
+input:checked ~ .fullsize{
+    display: block;
+    opacity: 1;
+}
+
+
+.close{display: none}
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: Write code in the expanded style even if only one property.

File: submissions/Zihfred/google pop-up/style.css [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+input:checked ~ .pop-up-more{
+    opacity: 0;
+    display: none;
+}
+input:checked ~ .fullsize{
+    display: block;
+    opacity: 1;
+}
+
+
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: Remove unnecessary lines.

File: submissions/Zihfred/google pop-up/style.css [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    border: 1px solid transparent;
+}
+.pop-up-item:hover{
+    border: 1px solid lightgrey;
+}
+.pop-up-item__img{
+    width: 40px;
+
+}
+input[type='checkbox']{
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: For this cases better use class.

File: submissions/Zihfred/google pop-up/style.css [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    top: -9px;
+    right: -216px;
+    margin-left: 4px;
+    width: 10px;
+    height: 10px;
+    transform: rotate(45deg);
+    border-left: 1px solid rgba(0,0,0,.2);
+    border-top: 1px solid rgba(0,0,0,.2);
+}
+
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: Some blocks of code separated by empty line some - not. Better use an empty line to separate code blocks.

File: submissions/Zihfred/google pop-up/index.html [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        <div class="menu_item">
+            <a href="#"><p>Gmail</p></a>
+        </div>
+        <div class="menu_item">
+            <a href="#"><p>Images</p></a>
+        </div>
+
+        <div class="menu_item pop-up-click">
+            <input type="checkbox" class="popup-checkbox" id="check">
+            <label for="check"><i class="fas fa-th"></i></a></label>
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: `:focus` state is missing on menu icon. Also, images inside popup have missing focus style too.

File: submissions/Zihfred/google pop-up/index.html [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+                    </div>
+                    <div class="pop-up-item close">
+                        <img src="img/Без названия (2).png" class="pop-up-item__img" alt="">
+                        <p class="pop-up-item__text">Classrom</p>
+                    </div>
+                    <div class="pop-up-item close">
+                        <img src="img/Google_Earth_Logo.svg.png" class="pop-up-item__img" alt="">
+                        <p class="pop-up-item__text">Earth</p>
+                    </div>
+                    <label class="pop-up-more" for="more-checkbox">More</label>
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: Add `:focus` and `:hover` state on this element.

File: submissions/Zihfred/google pop-up/index.html [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+                    <div class="fullsizepop-up">
+                        Even more from Google
+                    </div>
+                </div>
+
+
+            </div>
+        </div>
+        <div class="menu_item bell">
+            <a href="#"><i class="fas fa-bell"></i></a>
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/index.html]: Fix `:focus` styles for this element and user avatar.

File: submissions/Zihfred/google pop-up/style.css [hunk 13]
```diff
@@ -86,75 +85,85 @@ a {
...
+    border: 1px solid transparent;
+}
+.pop_up-more:hover{
+    cursor: pointer;
+    border: 1px solid blue;
+}
+.pop_up-more:focus{
+    border: 1px solid lightskyblue;
+}
+#more-checkbox,.popup_checkbox{
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: Avoid styling by id.

File: submissions/Zihfred/google pop-up/style.css [hunk 14]
```diff
@@ -86,75 +85,85 @@ a {
...
     text-align: center;
     width: 100%;
     text-decoration: underline;
 }
+.bell:hover, .username:hover{
+    border: 2px solid lightskyblue;
+}
+.bell:focus, .username:focus{
+    border: 2px solid lightskyblue;
+}
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/google pop-up/style.css]: Add an empty line at the end of the file.

**Comment by A-Ostrovnyy** [comment]
Nice!
On your menu icon missing `:focus` state. Also fix `:focus` state on the bell icon and avatar block.
Your menu must be accessible from the keyboard. When menu icon on focus I press Space or Enter key and popup must open. Same story with More button.
All interactive elements should have `:hover` and `:focus`. 
Add `cursor: pointer;` to interactive elements, it helps users to recognize interactive elements.

**Comment by A-Ostrovnyy** [comment]
Menu icon still missing outline on `:focus` (checked on Mac OS and Windows).
Form still inaccessible from the keyboard.

**Comment by A-Ostrovnyy** [comment]
Did you update your demo?

**Comment by Zihfred** [comment]
> Did you update your demo?

yeah

**Comment by A-Ostrovnyy** [comment]
Works good, but an order of focused elements need to be fixed: when I open popup, and press Tab key focus must going to the first link(Docs) and then I am iterating links inside popup by Tab key. After iterating all links focus must go to the button More. In your case order is chaotic.

**Comment by Zihfred** [comment]
New PR

**Pull request closed without merging**